lmdeploy/vl/model/internvl.py

```diff
@@ -130,23 +130,42 @@ def build_model(self):
                 self.model_path)
             self._forward_func = self._forward
 
-    def _preprocess_v1_5(self, images: List[Image]):
+    def _preprocess_v1_5(self,
+                         images: List[Image],
+                         image_kwargs: List[Dict] = None):
+        if image_kwargs is not None:
+            assert len(images) == len(
+                image_kwargs), 'different length of images and image_kwargs'
+        else:
+            image_kwargs = [{}] * len(images)
+
+        image_res = {
+            'low': 6,
+            'medium': 12,
+            'high': 24,
```

Comment:
Where are these values from?

Reply:
Most of the time, users have a strong demand for input resolution control. We refer to the [OpenAI](https://platform.openai.com/docs/guides/vision/low-or-high-fidelity-image-understanding) interface to set some restrictions on the number of tiles for input image processing. These numbers align with the settings used when testing InternVL. In addition, adding restrictions can effectively constrain the total tokens of multi-image input to be less than the session length (usually 8192 tokens in internvl), although the case of extreme multi-images is not currently considered.